### ISSUE TITLE
How to load a trained model in the right way?

### ISSUE THREAD

**Issue by ficht74** [lukas-blecher/LaTeX-OCR#283]
I want to make predictions with pix2tex with my fine-tuned version, but it doesn't work to load the config and weights? I tried it like this of pix2tex -h tells and __main__.py checks?

!python pix2tex --config "/content/drive/MyDrive/models/LaTeX-OCR/my_model_path/LaTeX-OCR-tuned-cf-allnew/config.yaml" --checkpoint "/content/drive/MyDrive/models/LaTeX-OCR/my_model_path/LaTeX-OCR-tuned-cf-allnew/LaTeX-OCR-tuned-cf-allnew_e10_step4144.pth"

I get this error:
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/dist-packages/pix2tex/__main__.py", line 32, in <module>
    main()
  File "/usr/local/lib/python3.10/dist-packages/pix2tex/__main__.py", line 27, in main
    from .cli import main
ImportError: attempted relative import with no known parent package

How can I load the tuned model in a right way? Thanx

**Comment by lukas-blecher** [comment]
Are you running it in a notebook?
If so, then load it similarly to 4. of "using the model" on the README

If you want to try it from the console, you're missing a '-m' after python. The error you've reported happens when the console is in the root directory of the repository.

**Comment by ficht74** [comment]
Yes i tried both on notebook and from the console - both works now. You were absolutely right that I have forgotten the -m - sorry shame on me ;-(

But I have a strange behavior by doing the evaluation my test data. With your model I get better Bleu-Scores than with mine, although I get a better bleu score 0.8557 when training.

I did the evaluation in a colab notebook and create the models like this:

**Your model:** model = pix2tex.LatexOCR() -> Bleu-Score: 0,710

**My model:**   my_model.LatexOCR(config='LaTeX-OCR-tuned-cf-allnew/config.yaml', checkpoint='LaTeX-OCR-tuned-cf-allnew_e09_step4144.pth', no_cuda=True, no_resize=False -> Bleu-Score: 0,6396

I am confused and cannot figure out the difference? Is it the temperature setting different or did I forgot a parameter?
I also cannot see if the state_dict_model and the  image_resizer.pth are correctly loaded in my case?

_self.image_resizer.load_state_dict(torch.load(os.path.join(os.path.dirname(self.args.checkpoint), 'image_resizer.pth'), map_location=self.args.device))

It would be really create if you could me support once again ;-)

Thanx

**Comment by lukas-blecher** [comment]
That's one you have to figure out by yourself. The resizer will be loaded the same way in your case. The problem is more fundamental I reckon. 
My best guess would be data leakage

**Comment by ficht74** [comment]
In the meantime I have tried a bit with the checkpoints and temperature parameters and found a setting that outperforms the pre-trained model, as I had expected. A data leakage is always possible, because a 100% check is usually not possible, but I would rather exclude it here, because I have already trained other models with the same data. But what was the Bond movie called "Never say never" ;-)

But if everything would always work immediately, it would be boring. 
Thanks again for your support!
Cheers Christian
